src/System.CommandLine/Parsing/SymbolLookupByName.cs

```diff
@@ -0,0 +1,175 @@
+                    return true;
+                }
+            }
+
+            if (skipAncestors)
+            {
+                break;
+            }
+        }
+
+        errorMessage = $"Requested symbol {name} was not found.";
+        symbol = null;
+        parent = null;
+        return false;
+    }
+
+    /// <summary>
+    /// Gets the symbol with the requested name that appears nearest to the starting command, which defaults to the current or leaf command.
+    /// </summary>
+    /// <param name="name">The name to search for</param>
+    /// <param name="startCommand">The command to start searching up from, which defaults to the current command.</param>
+    /// <param name="skipAncestors">If true, only the starting command and no ancestors are searched.</param>
+    /// <param name="valuesOnly">If true, commands are ignored and only options and arguments are found.</param>
+    /// <returns>A tuple of the found symbol and its parent command. Throws if the name is not found.</returns>
+    /// <exception cref="InvalidOperationException">Thrown if the name is not found.</exception>
+    public (CliSymbol symbol, CliCommand parent) GetSymbolAndParent(string name, CliCommand? startCommand = null, bool skipAncestors = false, bool valuesOnly = false)
+        => TryGetSymbolAndParentInternal(name, out var symbol, out var parent, out var errorMessage, startCommand, skipAncestors, valuesOnly)
+            ? (symbol, parent)
+            : throw new InvalidOperationException(errorMessage);
+
+
+    /// <summary>
+    /// Returns true if the symbol is found, and provides the symbols as the `out` symbol parameter.
+    /// </summary>
+    /// <param name="name">The name to search for</param>
+    /// <param name="symbol">An out parameter to receive the symbol if it is found.</param>
+    /// <param name="startCommand">The command to start searching up from, which defaults to the current command.</param>
+    /// <param name="skipAncestors">If true, only the starting command and no ancestors are searched.</param>
+    /// <param name="valuesOnly">If true, commands are ignored and only options and arguments are found.</param>
+    /// <returns>True if a symbol with the requested name is found</returns>
+    public bool TryGetSymbol(string name, [NotNullWhen(true)] out CliSymbol? symbol, CliCommand? startCommand = null, bool skipAncestors = false, bool valuesOnly = false)
+        => TryGetSymbolAndParentInternal(name, out symbol, out var _, out var _, startCommand, skipAncestors, valuesOnly);
+
+
+    private IEnumerable<CommandCache>? GetCommandCachesToUse(CliCommand currentCommand)
+    {
+        if (cache[0].Command == currentCommand)
+        {
+            return cache;
+        }
+        for (int i = 1; i < cache.Count; i++) // we tested for 0 earlier
+        {
+            if (cache[i].Command == currentCommand)
+            {
+                return cache.Skip(i);
+            }
+        }
+        return null;
```

Comment:
```suggestion
        int index = cache.FindIndex(c => c.Command == currentCommand);
        if (index == -1)
        {
            return null;
        }
        return cache.Skip(index);
```
I haven't found where this is used Could we do something with `.FindIndex` here instead of a for loop? It seems like it may be slightly easier to read and a bit simpler, and I'm not sure the short circuit at the top is saving much in terms of execution time, but if we think it is we could still keep that?